src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -108,4 +111,25 @@ private static CredentialsMatcher githubScanCredentialsMatcher() {
 
     private Connector() {}
 
+    /**
+     * Fail immediately and throw a customized exception.
+     */
+    public static final RateLimitHandler CUSTOMIZED = new RateLimitHandler() {
```

Comment:
@KostyaSha Indeed, there is another one `RateLimitHandler.FAIL` but throw a generic `IOException`. This implementation (`CUSTOMIZED`) is more flexible and I can do things like this:

```
Proposing build-pipeline-plugin
Connecting to GitHub using anonymous access
Looking up recena/build-pipeline-plugin

  Getting remote branches...

    Checking branch JENKINS-31746
      Jenkinsfile does not exist in this branch
    Does not meet criteria


    Checking branch cleanup
      Jenkinsfile does not exist in this branch
    Does not meet criteria


    Checking branch master
      Jenkinsfile does not exist in this branch
    Does not meet criteria


    Checking branch recovery
      Jenkinsfile does not exist in this branch
    Does not meet criteria


    Checking branch retry-param-and-security-fixes
      Jenkinsfile does not exist in this branch
    Does not meet criteria


    Checking branch revert-66-master
      Jenkinsfile does not exist in this branch
    Does not meet criteria

  0 branches were processed

Done examining recena/build-pipeline-plugin

Proposing cartodb
Connecting to GitHub using anonymous access
Looking up recena/cartodb

  Getting remote branches...

GitHub API rate limit exceeded

Aborted
Finished: ABORTED
```

In this PR I'm improving details related to UX.